### PR TITLE
Azure blob egress extension archives

### PR DESCRIPTION
**Pull request by jander-msft**
###### Summary

Create product and symbols archives for the Azure Blog Egress extension.

### CODE REVIEW

File: eng/AzureBlobStorageProjectToPublish.props [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+    <AzureBlobStoragePublishTargetFramework>$(LatestTargetFramework)</AzureBlobStoragePublishTargetFramework>
+    <AzureBlobStoragePublishTargetFramework Condition="'$(TargetFramework)' != ''">$(TargetFramework)</AzureBlobStoragePublishTargetFramework>
+    <AzureBlobStoragePublishPath>$(ArtifactsDir)pub\dotnet-monitor-egress-azureblobstorage\$(Configuration)\$(AzureBlobStoragePublishTargetFramework)\$(PackageRid)\</AzureBlobStoragePublishPath>
+  </PropertyGroup>
+  <ItemGroup>
+    <ProjectToPublish Include="$(RepoRoot)src\Extensions\AzureBlobStorage\AzureBlobStorage.csproj">
+      <AdditionalProperties>TargetFramework=$(AzureBlobStoragePublishTargetFramework);RuntimeIdentifier=$(PackageRid);PublishDir=$(AzureBlobStoragePublishPath)</AdditionalProperties>
+    </ProjectToPublish>
+  </ItemGroup>
+</Project>
```
Comment (jander-msft) [inline eng/AzureBlobStorageProjectToPublish.props:12]:
```suggestion
</Project>

```

File: src/archives/AzureBlobStorage.props [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<Project>
+  <!-- Import ProjectToPublish items -->
+  <Import Project="$(RepositoryEngineeringDir)AzureBlobStorageProjectToPublish.props" />
+  <PropertyGroup>
+    <ArchiveContentRootPath>$(AzureBlobStoragePublishPath)</ArchiveContentRootPath>
+  </PropertyGroup>
+  <!-- These items are included in addition to those from publishing the dotnet-monitor project. -->
```
Comment (jander-msft) [inline src/archives/AzureBlobStorage.props:7]:
```suggestion
  <!-- These items are included in addition to those from publishing the AzureBlobStorage project. -->
```

File: eng/Build.props [hunk 3]
```diff
@@ -1,6 +1,12 @@
 <Project>
   <Import Project="$(MSBuildThisFileDirectory)Common.props" />
   <ItemGroup Condition="'$(CreateArchives)' == 'true'">
+    <ProjectToBuild Include="$(RepoRoot)src\archives\pkgs\AzureBlobStorage\AzureBlobStorage-archive.proj">
```
Comment (wiktork) [inline eng/Build.props:4]: Long term: sharing common binaries across rids/architectures

**Review by wiktork — APPROVED** [review]

File: eng/Build.props (hunk 3 above)
Comment (jander-msft) [reply eng/Build.props:4]: As recently discussed, sharing the application binaries between multiple platform publishes is not really a supported scenario for .NET projects. I understand the intent here to minimize the payload of the extension.

One way to mitigate this is to be able to produce an extension as platform neutral and have `dotnet monitor` launch it using `dotnet`.

**Pull request merged**